music21/metadata/__init__.py

```diff
@@ -308,48 +743,281 @@ def all(self, skipContributors=False):
-                match = workId
-                break
-        if match is None:
-            raise AttributeError(f'object has no attribute: {name}')
-        result = self._workIds[match]
-        # always return string representation for now
-        return str(result)
+        '''
+        Utility attribute access for all uniqueNames, grandfathered workIds,
+        and grandfathered workId abbreviations.  Many grandfathered workIds
+        have explicit property definitions, so they won't end up here.
 
-    # PUBLIC METHODS #
+        These always return str or None.  If there is more than one item
+        for a particular name, we will try to summarize or list them all
+        in one returned string.
 
-    @staticmethod
-    def abbreviationToWorkId(abbreviation):
-        '''Get work id abbreviations.
+        If name is not a valid attribute (uniqueName, grandfathered workId,
+        or grandfathered workId abbreviation), then AttributeError is raised.
 
-        >>> metadata.Metadata.abbreviationToWorkId('otl')
-        'title'
+        >>> md = metadata.Metadata()
+        >>> md.description = metadata.Text('A description set via uniqueName', language='en')
+        >>> md.dedication = 'A dedication set via grandfathered workId'
+        >>> md.otl = metadata.Text('A title set via grandfathered workId abbreviation')
+        >>> md.description
+        'A description set via uniqueName'
+        >>> md.dedicatedTo  # the uniqueName for 'dedicated' workId
+        'A dedication set via grandfathered workId'
+        >>> md.title  # the uniqueName for 'otl' abbreviation
+        'A title set via grandfathered workId abbreviation'
+        >>> md.add('description', 'A description set via md.add()')
+        >>> md.description
+        'A description set via uniqueName, A description set via md.add()'
+        '''
 
-        >>> for work_id in metadata.Metadata.workIdAbbreviationDict:
-        ...    result = metadata.Metadata.abbreviationToWorkId(work_id)
+        # __getattr__ is the call of last resort after looking for bare
+        # attributes and property methods, so __getattr__ won't be called
+        # for self.software (bare attribute), or for self.composer (property),
+        # even though we certainly could handle those here. This is why we
+        # don't have to handle .composers/librettists/lyricists or
+        # .fileFormat/filePath/fileNumber here, like we do in __setattr__
+        # (which is the only call you get if you implement it, so it has
+        # to handle everything).
+        return self._getSingularAttribute(name)
+
+    def __setattr__(self, name: str, value: t.Any):
+        '''
+        Utility attribute setter for all uniqueNames, grandfathered workIds,
+        and grandfathered workId abbreviations.
 
+        These can take any single value, and will convert to the appropriate
+        internal valueType.  They will raise ValueError for iterables (other
+        than str).
```

Comment:
examples or point to `__getattr__` for examples.

Note that `__setattr__` and `__getattr__` and `__getitem__` and `__setitem__` are not, by default currently included in the music21 docs -- they probably need to be explicitly put in DOC_ATTR